src/data/error.rs

```diff
@@ -275,6 +275,43 @@ pub enum CppError {
     __Nonexhaustive,
 }
 
+#[derive(Clone, Copy, PartialEq, Eq, Debug)]
+pub enum Radix {
+    Binary,
+    Octal,
+    Decimal,
+    Hexadecimal,
+}
+
+use std::fmt;
+impl fmt::Display for Radix {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        write!(
```

Comment:
I don't like how rustfmt formatted this, maybe assign to a temporary variable first?

Reply:
How about a write macro in each branch of the match?

Reply:
Sure, that works too, it would just be a little more code.